presto-main/src/test/java/com/facebook/presto/operator/TestHashJoinOperator.java

```diff
@@ -1036,13 +1157,19 @@ public SingleStreamSpiller create(List<Type> types, SpillContext spillContext, L
                 public ListenableFuture<?> spill(Iterator<Page> pageIterator)
                 {
                     checkState(writing, "writing already finished");
+                    if (failSpill) {
+                        return immediateFailedFuture(new PrestoException(GENERIC_INTERNAL_ERROR, "Spill failed"));
+                    }
                     Iterators.addAll(spills, pageIterator);
                     return immediateFuture(null);
                 }
 
                 @Override
                 public Iterator<Page> getSpilledPages()
                 {
+                    if (failUnspill) {
+                        throw new PrestoException(GENERIC_INTERNAL_ERROR, "Unspill failed");
```

Comment:
You could also test a case when iterator is returned successfully, but fails during e.g. `hasNext()`